src/shell/element/surface.rs

```diff
@@ -73,6 +73,22 @@ impl From<X11Surface> for CosmicSurface {
     }
 }
 
+impl PartialEq<WlSurface> for CosmicSurface {
+    fn eq(&self, other: &WlSurface) -> bool {
+        self.wl_surface().map(|s| &s == other).unwrap_or(false)
+    }
+}
+
+impl PartialEq<X11Surface> for CosmicSurface {
+    fn eq(&self, other: &X11Surface) -> bool {
+        if self.0.is_x11() {
+            self.x11_surface().map(|s| s == other).unwrap_or(false)
```

Comment:
Calling `is_x11` and `x11_surface` seems redundant. `x11_surface()` should return `Some` if and only if `is_x11()`.

This could also use `map_or`.